backend/src/main/java/com/benine/backend/database/MySQLDatabase.java

```diff
@@ -232,6 +220,97 @@ public void addCamera(int id, String macAddress) {
   }
 
   @Override
+  public void checkCameras() throws SQLException {
+    ArrayList<Camera> cameras = ServerController.getInstance().getCameraController().getCameras();
+    ArrayList<String> macs = new ArrayList<String>();
+    ResultSet resultset = null;
+    Statement statement = null;
+    try {
+      statement = connection.createStatement();
+      String sql = "SELECT ID, MACAddress FROM camera";
+      resultset = statement.executeQuery(sql);
+      statement.close();
+      resultset.close();
+      checkOldCameras(resultset, cameras, macs);
+      checkNewCameras(cameras, macs);
+    } catch (SQLException e) {
+      e.printStackTrace();
+      getLogger().log("Cameras could not be gotten from database.", LogEvent.Type.CRITICAL);
+    } catch (CameraConnectionException e) {
+      e.printStackTrace();
+    } finally {
+      if (statement != null) {
+        statement.close();
+        resultset.close();
+      }
+    }
+  }
+
+  /**
+   * Cheks if there are cameras in the database to be deleted.
+   * @param result The resultset from the query
+   * @param cameras The cameras
+   * @param macs The MACAddresses of the cameras in the database
+   * @throws SQLException No right connection to the database
+   * @throws CameraConnectionException Not able to connect to the camera
+   */
+  public void checkOldCameras(ResultSet result, ArrayList<Camera> cameras, ArrayList<String> macs)
+      throws SQLException, CameraConnectionException {
+    while (result.next()) {
+      boolean contains = false;
+      String mac = result.getString("MACAddress");
+      macs.add(mac);
+      for (Camera camera : cameras) {
+        if (camera.getMacAddress().equals(mac)) {
+          contains = true;
+          break;
+        }
+      }
+      if (!contains) {
+        deleteCamera(result.getInt("ID"));
+      }
+    }
+  }
+
+  /**
+   * Checks if there are new cameras to be added to the database.
+   * @param cameras The cameras
+   * @param macs The MACAddresses of the cameras in the database
+   * @throws CameraConnectionException Not able to connect to the camera
+   */
+  public void checkNewCameras(ArrayList<Camera> cameras, ArrayList<String> macs)
+      throws CameraConnectionException {
+    boolean contains = false;
+    for (Camera camera : cameras) {
+      for (String mac : macs) {
+        if (mac.equals(camera.getMacAddress())) {
+          contains = true;
+          break;
+        }
+      }
+      if (!contains) {
+        addCamera(camera.getId(), camera.getMacAddress());
+      }
+    }
+  }
+
+  @Override
+  public void deleteCamera(int cameraID) throws SQLException {
+    Statement statement = connection.createStatement();
+    try {
+      String sql = "DELETE FROM presets WHERE camera_ID = " + cameraID;
+      statement.executeUpdate(sql);
+      sql = "DELETE FROM camera WHERE ID = " + cameraID;
+      statement.executeUpdate(sql);
+      statement.close();
```

Comment:
This can be removed since it will be called in the finally block anyway. Calling close two times might result in weird behaviour.